src/main/java/net/minecraftforge/fml/common/registry/GameData.java

```diff
@@ -200,19 +200,39 @@ private ResourceLocation addPrefix(String name)
         return new ResourceLocation(prefix, name);
     }
 
+    @SuppressWarnings("unchecked")
     void registerSubstitutionAlias(String name, GameRegistry.Type type, Object toReplace) throws ExistingSubstitutionException
     {
         ResourceLocation nameToSubstitute = new ResourceLocation(name);
-        final BiMap<Block, Item> blockItemMap = getBlockItemMap();
-        if (type == GameRegistry.Type.BLOCK)
+        FMLControlledNamespacedRegistry registry = null;
+        switch (type)
```

Comment:
You mean as field of GameDate which is populated in the constructor? 
Yup that would be cleaner, but I did not want to change to "much". If it would be accepted, I can change that.